crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -20,19 +20,39 @@ module Api
   class Upgrade < Tableless
     def status
       {
-        crowbar: crowbar_upgrade_status,
-        checks: check
-      }
+        crowbar: crowbar_upgrade_status
+      }.merge!(check)
     end
 
     def check
-      {
-        sanity_checks: sanity_checks,
-        maintenance_updates_installed: maintenance_updates_installed?,
-        clusters_healthy: clusters_healthy?,
-        compute_resources_available: compute_resources_available?,
-        ceph_healthy: ceph_healthy?
-      }
+      {}.tap do |ret|
+        ret[:checks] = {}
+        ret[:checks][:network_checks] = {
+          required: true,
+          passed: network_checks.empty?,
+          errors: sanity_check_errors
+        }
+        ret[:checks][:maintenance_updates_installed] = {
+          required: true,
+          passed: maintenance_updates_status.empty?,
+          errors: maintenance_updates_check_errors
+        }
+        ret[:checks][:compute_resources_available] = {
+          required: false,
+          passed: compute_resources_available?,
+          errors: compute_resources_check_errors
+        }
+        ret[:checks][:ceph_healthy] = {
+          required: true,
+          passed: ceph_healthy?,
+          errors: ceph_health_check_errors
+        } if Api::Crowbar.new.addons.include?("ceph")
```

Comment:
Why do we need to instantiate an object here? Is there really any useful state inside that object?  If not, shouldn't it be a class method?